tools/okparser/src/parser.py

```diff
@@ -0,0 +1,85 @@
+import re
+from pathlib import Path
+
+import requests
+import typer
+import yaml
+
+from tools.okparser.src import utils
+
+__REQUIRED_FIELDS__ = frozenset({"bom", "tool-list"})
+
+
+def get_wiki_data(items: list[str]) -> list[dict[str, str]]:
+    """search the wikidata api for items.
+
+    Args:
+        items: list of items(bom items, tools) to be searched.
+    Returns:
+        A list of search descriptions for each item.
+    """
+    item_list = []
+    with typer.progressbar(items, color=True) as progress:
+        for item in progress:
+            item_dict = {"identifier": "", "description": item.strip(), "link": ""}
+            # look for 'or' keyword in text and search each item
+            sub_items = item.split("or")
+            for sub_item in sub_items:
+                response = requests.get(utils.get_url(sub_item)).json()
+                if response["search"]:
+                    # get the first result from the search results for now
+                    item_dict["identifier"] = response["search"][0]["id"]
+                    item_dict["link"] = re.sub(
+                        "//www.", "https://", response["search"][0]["url"]
+                    )
+                    break
+                utils.console.print(f"\t[red]couldn't find wikidata for {sub_item}")
+            item_list.append(item_dict)
+    return item_list
+
+
+class Okh:
+    # list of bom atoms with wiki data
+    bom_atoms: list[dict[str, str]]
+    # list of tools with wikidata
+    tool_list_atoms: list[dict[str, str]]
+    # source_path of yaml file
+    source_path: Path
```

Comment:
Note sure the Okh file should store it's source/destintation file names as properties. Maybe this is just me, but I would expect a static `open` method that takes a file path parameter and then a save instance method (as you have below) that takes a file path parameter as well. This class should be responsible for reading/writing the OKH from/to a specified location, but I think the file location management logic should be separated out